Is accepting Fn instead of FnMut intentional?

I've been trying to write the following code:

```rust
fn main() {
    // allocate output buffer once and reuse it, this is faster
    let mut out: Vec<u8> = Vec::new();
    fuzz!(|data: &[u8]| {
        out.truncate(0);
        let mut decoder = GzDecoder::new(data);
        decoder.read_to_end(&mut out);
    });
}
```
However, this fails:
```
error[E0596]: cannot borrow `out` as mutable, as it is a captured variable in a `Fn` closure
  --> src/main.rs:12:9
   |
12 |         out.truncate(0);
   |         ^^^ cannot borrow as mutable
   |
help: consider changing this to accept closures that implement `FnMut`
  --> src/main.rs:10:5
   |
10 | /     fuzz!(|data: &[u8]| {
11 | |         let mut decoder = GzDecoder::new(data);
12 | |         out.truncate(0);
13 | |         decoder.read_to_end(&mut out);
14 | |     });
   | |_______^
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
```
I have to write the following inefficient code instead:
```rust
fn main() {
    fuzz!(|data: &[u8]| {
        let mut out: Vec<u8> = Vec::new();
        let mut decoder = GzDecoder::new(data);
        decoder.read_to_end(&mut out);
    });
}
```
Is the `Fn` bound and not `FnMut` intentional?

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -122,9 +122,9 @@ extern "C" {
 /// });
 /// # }
 /// ```
-pub fn fuzz<F>(hook: bool, closure: F)
+pub fn fuzz<F>(hook: bool, mut closure: F)
 where
-    F: Fn(&[u8]) + std::panic::RefUnwindSafe,
+    F: FnMut(&[u8]) + std::panic::RefUnwindSafe,
 {
     // this marker strings needs to be in the produced executable for
     // afl-fuzz to detect `persistent mode` and `defered mode`
@@ -139,10 +139,10 @@ where
     // unsafe { asm!("" : : "r"(&DEFERED_MARKER)) };
 
     if hook {
-      // sets panic hook to abort
-      std::panic::set_hook(Box::new(|_| {
-          std::process::abort();
-      }));
+        // sets panic hook to abort
+        std::panic::set_hook(Box::new(|_| {
+            std::process::abort();
+        }));
     }
 
     let mut input = vec![];
@@ -161,9 +161,9 @@ where
         // the panic hook.
         // If so, the fuzzer will be unable to tell different bugs appart and you will
         // only be able to find one bug at a time before fixing it to then find a new one.
-        let did_panic = std::panic::catch_unwind(|| {
+        let did_panic = std::panic::catch_unwind(panic::AssertUnwindSafe(|| {
             closure(&input);
-        })
+        }))
         .is_err();
 
         if did_panic {

